### PR TITLE
Add tests vs pennylane rc branch

### PR DESCRIPTION
**Pull request by lillian542**
**Context:**
We have this test matrix [here](https://github.com/PennyLaneAI/plugin-test-matrix#feature-freeze-testing-matrix) that runs the tests every day for each plugin against the release candidate branch of Pennylane. It's a way to make sure nothing is broken between the release branch and the other relevant repositories as we prepare for the release. Since Catalyst is part of the release now, it should also be included.

**Description of the Change:**
Added an additional install option for pennylane in the main comparison workflow for Pennylane, Catalyst and Lightning. It's currently set to the previous release candidate branch. Also added a workflow with `pennylane: release-candidate` that runs this. 

**Benefits:**
We can add this workflow to the matrix linked above, and easily monitor compatibility with the release candidate branch and catalyst throughout feature freeze.

**Possible Drawbacks:**
We will need to update the release candidate branch name in the workflow on the Monday of feature freeze every release cycle. Also it adds another CI workflow.

### CODE REVIEW

File: .github/workflows/CPL_latest-rc-latest.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: Check CPL latest/latest/latest
```
Comment (lillian542) [inline .github/workflows/CPL_latest-rc-latest.yaml:1]:
```suggestion
name: Check CPL latest/rc/latest
```

**Comment by lillian542** [comment]
[sc-49455]

File: .github/workflows/CPL_latest-rc-latest.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: Check CPL latest/rc/latest
+
+on:
+  schedule:
+    - cron: "20 10 * * 1-5"
```
Comment (dime10) [inline .github/workflows/CPL_latest-rc-latest.yaml:5]: Do you have any preference on when and/or how frequent this action should run?

**Review by dime10 — APPROVED** [review]
Looks great to me, thank you for the contribution 🥳 🥳 

Once this is merged we should be able to generate a badge for it.

**Comment by dime10** [comment]
Another thought I just had, should this action be disabled outside of feature freeze? (not sure how you are handling it on the PL side)

File: .github/workflows/CPL_latest-rc-latest.yaml (hunk 2 above)
Comment (lillian542) [reply .github/workflows/CPL_latest-rc-latest.yaml:5]: I can set it to match the others for the RC branch test matrix - they usually run every day at midnight. 

If there is a way to easily disable outside of feature freeze without updating the PR before and after each one, I don't know of it. The other RC branch tests just run every day no matter whether its feature freeze or not 🤷‍♀️

Comment (lillian542) [inline .github/workflows/CPL_latest-rc-latest.yaml:5]:
```suggestion
    - cron: "0 0 * * *"
```

File: .github/workflows/CPL_latest-rc-latest.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: Check CPL latest/rc/latest
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+
+jobs:
+  build-config:
+    name: CPL latest/latest/latest
```
Comment (lillian542) [inline .github/workflows/CPL_latest-rc-latest.yaml:10]:
```suggestion
    name: CPL latest/rc/latest
```

File: .github/workflows/CPL_latest-rc-latest.yaml (hunk 2 above)
Comment (dime10) [reply .github/workflows/CPL_latest-rc-latest.yaml]: > The other RC branch tests just run every day no matter whether its feature freeze or not 🤷‍♀️

Actions can be enabled/disabled pretty easily from the actions tab :)
image

Comment (lillian542) [reply .github/workflows/CPL_latest-rc-latest.yaml]: Ah, good to know, thanks!

**Pull request merged**